app/provider/app.go

```diff
@@ -376,7 +376,7 @@ func New(
 		stakingtypes.NewMultiStakingHooks(
 			app.DistrKeeper.Hooks(),
 			app.SlashingKeeper.Hooks(),
-			app.ProviderKeeper.Hooks(),
+			app.ProviderKeeper.Hooks(app.GovKeeper),
```

Comment:
As far as I can tell this is not very conventional, but on the other hand, adding the `govKeeper` to the `providerKeeper` is even more ugly, because the `govKeeper` needs the `providerKeeper` for the proposal routes. So we would have to write something like (pseudo code): 

```go
providerKeeper := providerKeeper.NewKeeper( ...  govKeeper ...) // govKeeper is nil for now
govKeeper := govKeeper.NewKeeper( ... providerKeeper ... )
providerKeeper.SetGovKeeper( govKeeper ) // finally set a non-nil govKeeper
```

Reply:
I see one benefit of having an additional parameter in `Hooks()`: it breaks compilation after upgrade. Typically when gaia will upgrade its interchain-security dependency, the fix will be obvious, while on the other hand, having to call `SetGovKeeper` can be easily forgotten.
Is there any validation we could setup to ensure `SetGovKeeper` is actually called ? A little bit like `keeper.mustValidateField` but for setters.